File: app/src/main/java/com/stedi/randomimagegenerator/app/presenter/interfaces/ChooseSaveOptionsPresenter.java
package com.stedi.randomimagegenerator.app.presenter.interfaces;

import android.graphics.Bitmap;
import android.support.annotation.NonNull;

import com.stedi.randomimagegenerator.app.presenter.interfaces.core.Presenter;
import com.stedi.randomimagegenerator.app.presenter.interfaces.core.UI;

public interface ChooseSaveOptionsPresenter extends Presenter<ChooseSaveOptionsPresenter.UIImpl> {
    void setQualityFormat(@NonNull Bitmap.CompressFormat format);

    void setQualityValue(int value);

    void getData();

    interface UIImpl extends UI {
        void showQualityFormat(@NonNull Bitmap.CompressFormat format);

        void showQualityValue(int value);

        void showSaveFolder(@NonNull String path);

        void onIncorrectQualityValue();
    }
}

File: app/src/main/java/com/stedi/randomimagegenerator/app/presenter/interfaces/ChooseSaveOptionsPresenter.kt
package com.stedi.randomimagegenerator.app.presenter.interfaces

import android.graphics.Bitmap
import com.stedi.randomimagegenerator.app.presenter.interfaces.core.Presenter
import com.stedi.randomimagegenerator.app.presenter.interfaces.core.UI

interface ChooseSaveOptionsPresenter : Presenter<ChooseSaveOptionsPresenter.UIImpl> {
    fun setQualityFormat(format: Bitmap.CompressFormat)

    fun setQualityValue(value: Int)

    fun getValues()

    interface UIImpl : UI {
        fun showQualityFormat(format: Bitmap.CompressFormat)

        fun showQualityValue(value: Int)

        fun onIncorrectQualityValue()
    }
}